examples/producer_example/producer_example.go

```diff
@@ -45,14 +45,17 @@ func main() {
 
 	// Optional delivery channel, if not specified the Producer object's
 	// .Events channel is used.
-	deliveryChan := make(chan kafka.Event)
+	deliveryChan := make(chan kafka.Event, 1)
 
 	value := "Hello Go!"
 	err = p.Produce(&kafka.Message{
 		TopicPartition: kafka.TopicPartition{Topic: &topic, Partition: kafka.PartitionAny},
 		Value:          []byte(value),
 		Headers:        []kafka.Header{{Key: "myTestHeader", Value: []byte("header values are binary")}},
 	}, deliveryChan)
+	if err != nil {
+		fmt.Printf("Failed to produce the message: %s\n", err.Error())
```

Comment:
This should probably exit/return as well since the program will hang waiting on the deliveryChan if no message was produced.

Reply:
Done in 116c9559ba54e9381b7e6f5a5a64bf6632396208